### ISSUE TITLE
False positive for redefined function

### ISSUE THREAD

**Issue by capaizee** [PyCQA/pylint#4016]
Code works, false positive, but maybe no need or to hard to fix :)

Too many positional arguments for function call,
Unexpected keyword argument 'encoding' in function call

```
from flask import Flask
app = Flask(__name__)

@app.route("/")
def test():
    with open("page-99.html", encoding="utf-8") as fp:
        temp = fp

@app.route("/open")
def open():
    return "ok"

if __name__ == "__main__":
    app.run()
```

**Comment by capaizee** [comment]
Sorry when function called  its throws error so correct error message...
